### DISCUSSION TITLE
Map rotation

### DISCUSSION THREAD

**Discussion by alina26** [osmdroid/osmdroid#1967] [Q&A]
@spyhunter99 hi! I suppose that you can know how rotation in osmdroid happens as you write this rotation gesture logic. Let's begin with formulation of issue I'm trying to solve - I am implementing course up mode for navigation so I need to rotate map around user's location(but not around center as it is done when calling mapView.setMapOrientation(degrees)). But I noticed that rotation around arbitrary point works perfect when we do rotation gesture so I thought that I can look for this logic somewhere here. But analyzing RotationGestureDetector and RotationGestureOverlay I didn't find nothing except mapView.setMapOrientation(degrees). So the question is - how RotationGestureOverlay rotates around touch position and not center?

**Comment by spyhunter99** [comment]
hmm i don't remember writing that class. Not sure what you're trying to do, but i think what you want is this...
https://github.com/osmdroid/osmdroid/blob/master/OpenStreetMapViewer/src/main/java/org/osmdroid/samplefragments/location/SampleHeadingCompassUp.java

other than that, i'm not sure i know the answer, sorry

**Reply by alina26** [reply]
it's not exactly what I'm looking for but anyway thanks a lot for help :)
